source/administration/changelog.md

```diff
@@ -4,9 +4,269 @@ This changelog summarizes updates to [Mattermost Team Edition](http://www.matter
 
 Also see [changelog in progress](http://bit.ly/2nK3cVf) for the next release.
 
-## Release v3.10.0
+## Release v4.0.0
 
-Release Date: June 16, 2017
+Release Date: July 16, 2017
+
+### Security Update
+
+- Mattermost v4.0.0 contains multiple security fixes. [Upgrading](http://docs.mattermost.com/administration/upgrade.html) is highly recommended. Details will be posted on our [security updates page](https://about.mattermost.com/security-updates/) 14 days after release as per the [Mattermost Responsible Disclosure Policy](https://www.mattermost.org/responsible-disclosure-policy/).
+
+### Highlights
+
+#### Native iOS and Android Apps
+- Second generation mobile apps released for [iOS](https://itunes.apple.com/us/app/mattermost/id1257222717?mt=8) and [Android](https://play.google.com/store/apps/details?id=com.mattermost.rn).
+- The apps are [EMM compatible starting with BlackBerry Dynamics](https://about.mattermost.com/mattermost-2nd-gen-mobile-apps-released-emm-compatible-starting-with-blackberry-dynamics/).
+
+#### Updated Web User Interface
+- Updated the appearance of channel header and channel sidebar in the web user interface.
+- Added a new default theme, "Mattermost". To try it, go to **Account Settings > Display > Theme**.
+
+#### Emoji Picker
+- The emoji picker offers quick access to emoji when composing messages or adding reactions.
+- Promoted from Beta, and enabled to all users by default.
+
+#### Languages
+- Added Italian translations to the user interface.
+
+#### API v4 (Stable Release)
+- Mattermost webapp moved to API v4 endpoints, which allow for more powerful integrations and server interaction.
+- API v3 endpoints are supported until January 16, 2018. To learn more about migrating to APIv4 endpoints, [see https://api.mattermost.com/](https://api.mattermost.com/).
+
+#### High Availability ([Enterprise Edition E20](https://about.mattermost.com/pricing/))
+- Mattermost servers are dynamically added and removed based on discovery and their cluster name using the hashicorp memberlist. 
+- Added support for experimental gossip protocol, where the server will attempt to communicate via the gossip protocol over the gossip port.
+
+### Improvements
+
+#### Web User Interface
+- Adjusted post spacing to be consistent across Markdown formatting, replies and consecutive posts.
+- On hover colour for pin and channel member icons now consistent with flag and recent mentions icons.
+- Emojis are now vertically aligned in post view.
+- Channel name, header and purpose now update in real time for all users.
+- For reply threads in the center channel, the "Commented on" phrase now respects the teammate name display setting.
```

Comment:
Yes, but it still respects it. I'll change to `teammate name display config setting`